File: second_homework/shapes/tetrahedron.cpp
#include "tetrahedron.h"
#include "shape.h"
#include "cmath"

tetrahedron::tetrahedron(int side, double density) : shape(density) {
    this->side = side;
}

double tetrahedron::calculate_surface_area() {
    return this->side * this->side * sqrt(3);
}

void tetrahedron::to_string(FILE *outputFile) {
    fprintf(outputFile, "[TETRAHEDRON] Side = %d Surface Area = %lf Density = %lf\n",
            side, calculate_surface_area(), density);
}

void tetrahedron::to_string() {
    printf("[TETRAHEDRON] Side = %d Surface Area = %lf Density = %lf\n",
           side, calculate_surface_area(), density);
}

File: third_homework/shapes/tetrahedron.py
import math

from shapes import shape


class Tetrahedron(shape.Shape):
    def __init__(self, side, density):
        super(Tetrahedron, self).__init__(density)
        self.side = side

    @property
    def surface_area(self):
        return self.side ** 2 * math.sqrt(3)

    def __str__(self):
        return f'[TETRAHEDRON] side = {self.side} | density = {self.density} | surface_area = {self.surface_area}'


__all__ = ['Tetrahedron']
